monitor.go

```diff
@@ -38,22 +39,24 @@ func (m *Monitor) Stop() {
 	m.setState(Stopped, nil)
 }
 
-// Register registers an observer, whose OnNotify method will be called any time there is a state change.  Does nothing
-// if the observer is already registered.
-func (m *Monitor) Register(o Observer) {
+// RegisterCallback registers a function which will be called any time there is a state change.  Returns a token that
+// can be used to deregister it later.
+func (m *Monitor) RegisterCallback(f func(Event)) Token {
 	m.Lock()
 	defer m.Unlock()
-	if m.observers == nil {
-		m.observers = make(map[Observer]struct{})
+	if m.callbacks == nil {
+		m.callbacks = make(map[Token]func(Event))
 	}
-	m.observers[o] = struct{}{}
+	token := Token(rand.Uint64())
```

Comment:
( minor ) could it be just an internal counter/index instead of the random one?

Reply:
I like that idea, but would you be able to prevent two different monitors from issuing the same token?

Reply:
I coded it this way so that there wouldn't be an overlap in token values between different monitors.

I think we can do both, though.  Let me update this.